pinot-core/src/test/java/org/apache/pinot/core/data/function/InbuiltFunctionEvaluatorTest.java

```diff
@@ -120,10 +128,27 @@ public void testStateSharedBetweenRowsForExecution()
     assertEquals(evaluator.evaluate(row), "test test test ");
   }
 
+  @Test
+  public void testExceptionDuringInbuiltFunctionEvaluator()
+      throws Exception {
+    String expression = "fromDateTime('2020-01-01T00:00:00Z', \"invalid_identifier\")";
+    InbuiltFunctionEvaluator evaluator = new InbuiltFunctionEvaluator(expression);
+    assertEquals(1, evaluator.getArguments().size());
+    GenericRow row = new GenericRow();
+    try {
+      evaluator.evaluate(row);
+      fail();
+    } catch (Exception e) {
+      assertTrue(e instanceof RuntimeException);
+      assertTrue(e.getCause() instanceof IllegalStateException);
+    }
+  }
+
   private static class MyFunc {
     String _baseString = "";
 
     String appendToStringAndReturn(String addedString) {
+      Preconditions.checkNotNull(addedString);
```

Comment:
Seems unrelated?

Reply:
yeah sorry I was planning to use MyFunc, will revert this